packages/liveblocks-core/src/client.ts

```diff
@@ -314,6 +315,26 @@ export type Client<U extends BaseUserMeta = DU, M extends BaseMetadata = DM> = {
    */
   logout(): void;
 
+  /**
+   * Advanced APIs related to the resolvers.
+   */
+  resolvers: {
+    /**
+     * Invalidate one or all users that were previously cached by `resolveUsers`.
+     */
+    invalidateUsers(userId?: string): void;
```

Comment:
Maybe instead of `userId?: string`, you could accept an array by default:

```suggestion
    invalidateUsers(userIds: string[]): void;
```

Perhaps even also let it take a predicate function, so they can implement the invalidation very precisely according to whatever logic makes sense to them:

```suggestion
    // All of these overloads would make sense
    invalidateUsers(): void;
    invalidateUsers(userIds: string[]): void;
    invalidateUsers(predicate: (user: User) => boolean): void;
```

This would allow these calls:

```ts
client.resolves.invalidateUsers();  // Invalidate all
client.resolves.invalidateUsers(["user1"]);  // Invalidate a single user
client.resolves.invalidateUsers(["user1", "user2"]);  // Invalidate multiple users
client.resolves.invalidateUsers((user) => user.info.tags.includes("Engineering"));
```

What do you think?

Reply:
Yes, love it! I'm going to implement that on the plane ✈️